FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/MessageDetailsFragment.kt

```diff
@@ -462,10 +467,49 @@ class MessageDetailsFragment : BaseFragment(), ProgressBehaviour, View.OnClickLi
 
   private fun updateMsgBody() {
     if (msgInfo != null) {
+      updatePgpBadges()
       updateMsgView()
     }
   }
 
+  private fun updatePgpBadges() {
+    msgInfo?.verificationResult?.let { verificationResult ->
+      val badges = mutableListOf<PgpBadgeListAdapter.PgpBadge>()
+
+      if (msgInfo?.encryptionType == MessageEncryptionType.ENCRYPTED) {
+        badges.add(PgpBadgeListAdapter.PgpBadge(PgpBadgeListAdapter.PgpBadge.Type.ENCRYPTED))
+      } else {
+        badges.add(PgpBadgeListAdapter.PgpBadge(PgpBadgeListAdapter.PgpBadge.Type.NOT_ENCRYPTED))
```

Comment:
On iOS we also have a third badge option: `decrypt error`. I think it would be worth adding sometime. (in the future)

Reply:
I've added an issue for that (#1609)